Blish HUD/GameServices/ModuleService.cs

```diff
@@ -317,20 +320,61 @@ public class ModuleService : GameService {
         }
 
         private void Gw2Instance_Gw2Started(object sender, EventArgs e) {
-            foreach (var module in _modules) {
-                if (module.State.Enabled) {
-                    module.TryEnable();
+            var resolvedDependencyOrder = SortByDependency(_modules, module => {
+                var dependencyModules = _modules.Where(m => {
+                    // Get all modules which have a dependency on the current module
+                    return module.Manifest?.Dependencies?.Any(dm => !dm.IsBlishHud && dm.Namespace == m.Manifest.Namespace) ?? false;
+                });
+
+                return dependencyModules;
+            }).ToList();
+
+            if (resolvedDependencyOrder != null) {
+                Logger.Debug($"Resolved dependency order: {string.Join(", ", resolvedDependencyOrder.Select(x => x.Manifest.Namespace).ToArray())}");
+
+                foreach (var module in resolvedDependencyOrder) {
+                    if (module.State.Enabled) {
+                        module.TryEnable();
+                    }
+                }
+            }else {
+                Logger.Error($"Could not resolve dependencies of any module.");
+            }
+        }
+
+        private static IEnumerable<ModuleManager> SortByDependency(IEnumerable<ModuleManager> source, Func<ModuleManager, IEnumerable<ModuleManager>> dependencies, bool throwOnCycle = false, bool logOnCycle = true) {
+            var sorted = new List<ModuleManager>();
+            var visited = new HashSet<ModuleManager>();
+
+            foreach (var item in source)
+                VisitDependency(item, visited, sorted, dependencies, throwOnCycle, logOnCycle);
+
+            return sorted;
+        }
+
+        private static void VisitDependency(ModuleManager item, HashSet<ModuleManager> visited, List<ModuleManager> sorted, Func<ModuleManager, IEnumerable<ModuleManager>> dependencies, bool throwOnCycle, bool logOnCycle) {
+            if (!visited.Contains(item)) {
+                visited.Add(item);
+
+                foreach (var dep in dependencies(item))
+                    VisitDependency(dep, visited, sorted, dependencies, throwOnCycle, logOnCycle);
+
+                sorted.Add(item);
+            } else {
+                if (!sorted.Contains(item)) {
+                    if (logOnCycle) Logger.Warn($"Cyclic dependency found: {item.Manifest.Namespace}");
+                    if (throwOnCycle) throw new Exception($"Cyclic dependency found: {item.Manifest.Namespace}");
```

Comment:
If this throws, what happens?

Reply:
I was undecided what to do with that. I will probably remove it as its not really needed.
I already tested it without the throw. It just does not load both modules